src/plugins_types.c

```diff
@@ -506,33 +505,34 @@ ly_type_validate_patterns(struct lysc_pattern **patterns, const char *str, size_
         }
 
         rc = pcre2_match(patterns[u]->code, (PCRE2_SPTR)str, str_len, 0, PCRE2_ANCHORED | PCRE2_ENDANCHORED, match_data, NULL);
-        if (rc == PCRE2_ERROR_NOMATCH) {
-            if (asprintf(&errmsg, "String \"%.*s\" does not conform to the pattern \"%s\".", (int)str_len, str, patterns[u]->expr) == -1) {
+        pcre2_match_data_free(match_data);
+
+        /* ERROR CHECK */
+        /* !(rc == PCRE2_ERROR_NOMATCH || rc >= 0) */
+        if ((rc != PCRE2_ERROR_NOMATCH) && (rc < 0)) {
+            PCRE2_UCHAR pcre2_errmsg[LY_PCRE2_MSG_LIMIT] = {0};
+            pcre2_get_error_message(rc, pcre2_errmsg, LY_PCRE2_MSG_LIMIT);
+
+            *err = ly_err_new(LY_LLERR, LY_ESYS, 0, strdup((const char *)pcre2_errmsg), NULL, NULL);
+            return LY_ESYS;
+        }
+        /* MATCH CHECK   */
+        else if (((rc == PCRE2_ERROR_NOMATCH) && (patterns[u]->inverted == 0)) ||
+                ((rc != PCRE2_ERROR_NOMATCH) && (patterns[u]->inverted == 1))) {
+            LY_ERR ret = 0;
+            const char *inverted = patterns[u]->inverted == 0 ? " " : " inverted ";
```

Comment:
Do you feel that it is necessary to keep it in a variable? I'm not strictly against it, but in such a simple case it seems to be better to put the conditional evaluation just as a parameter of the `asprintf()`.

Reply:
function asprintf take two lines already.  So I wanted expand that command as less as possible.

Reply:
so you have added the extra line somewhere else to hide a little more the string to be printed :)